graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/processor/OperationModelBuilderBase.java

```diff
@@ -90,16 +92,42 @@ protected OperationModelBuilderBase(
    */
   protected List<WhereConditionModel> buildWhereConditions(EntityModel entity)
       throws SkipException {
-    ImmutableList.Builder<WhereConditionModel> whereConditionsBuilder = ImmutableList.builder();
+    return buildConditionModel(
+        entity,
+        (inputValue, e) ->
+            new WhereConditionModelBuilder(inputValue, operationName, e, entities, context),
+        (value) -> DirectiveHelper.getDirective("cql_if", value).isPresent());
+  }
```

Comment:
This is a bit hard to follow (especially the fact that the predicate is exclusive).

Also it won't detect some invalid usages:
- a parameter is annotated with both `@cql_if` and `@cql_where`
- a parameter is annotated with `@cql_if` in an operation that doesn't support it (e.g. a SELECT).

Maybe we could have a single method that builds the two lists.

Reply:
Implemented in: https://github.com/stargate/stargate/pull/958/commits/26da160b1ef7d7b2c899fb047ca13d32e263785b

Reply:
> Is there a way to get the info if the operation is Select

You can check in `QueryModelBuilder`:
```java
    ConditionsModelBuilder.Conditions conditions = buildConditions(entity);
    if (!conditions.getIfConditions().isEmpty) {
      ... // fail
    }
    List<ConditionModel> whereConditions = conditions.getWhereConditions();
```

Or maybe pass an additional boolean parameter `allowIfConditions` to `ConditionsModelBuilder.build()`.